src/jvmMain/kotlin/fr/acinq/lightning/io/JvmTcpSocket.kt

```diff
@@ -189,10 +190,9 @@ internal actual object PlatformSocketBuilder : TcpSocket.Builder {
     }
 }
 
-fun tlsExceptionHandler(logger: Logger) = CoroutineExceptionHandler { _, throwable -> logger.error(throwable) { "TLS socket error: " } }
-
 /**
- * Using the IO dispatcher is a good practice, but the TLS internal coroutines sometimes throw exceptions
- * that are fatal on Android and aren't caught in the parent context, so we catch them explicitly.
+ * The TLS internal coroutines are launched in a background scope that doesn't let us do fine-grained supervision.
+ * They may throw exceptions when the socket is remotely closed, which crashes the application on Android.
+ * This should be fixed by https://github.com/ktorio/ktor/pull/3690, but for now we need to explicitly handle exceptions.
  */
-fun tlsContext(logger: Logger) = Dispatchers.IO + tlsExceptionHandler(logger)
+fun tlsContext(logger: Logger) = Dispatchers.IO + CoroutineExceptionHandler { _, throwable -> logger.error(throwable) { "TLS socket error: " } }
```

Comment:
Do we still need a `CoroutineExceptionHandler` here ?

Reply:
Yes, unfortunately we do until https://github.com/ktorio/ktor/pull/3690 ships, because:

- ktor's TLS stack will throw exceptions if the connection is remotely terminated
- ktor creates a separate background scope for the tls channels, which doesn't let us handle that exception anywhere else

Reply:
Is it possible to highlight this in one of our tests because I don't think it's used at all atm ? If we remove this `CoroutineExceptionHandler` they still pass

Reply:
No we can't, for the same reason explained in https://github.com/ktorio/ktor/pull/3690: the test framework swallows those exceptions.